src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -301,6 +303,7 @@ private bool TryValidateMethod()
         {
             // :method
             _methodText = HttpRequestHeaders.HeaderMethod.ToString();
+            HttpRequestHeaders.HeaderMethod = default; // Suppress pseduo headers from the public headers collection.
```

Comment:
We identified a bug with this approach in 6.0. On line 221 above, TryValidateMethod is skipped if the method was sent via static hpack index. That means we fail to remove it from the collection.

https://github.com/dotnet/aspnetcore/issues/38581 fixed this inadvertently by bulk clearing the set bits in the collection itself rather than clearing the individual properties.